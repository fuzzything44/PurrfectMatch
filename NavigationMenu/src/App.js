import React, { Component } from 'react';
import PurrfectMenu from './PurrfectMenu';

import logo from './logo.png';

class App extends Component {
  render() {
    let links = [
      { label: 'Success Stories', link: '#Success-Stories', active: true },
      { label: 'Contact Us', link: '#Contact-Us' },
    ];

    return (
      <div className="container center">
        <PurrfectMenu links={links} logo={logo} />
      </div>
    );
  }
}

export default App;
