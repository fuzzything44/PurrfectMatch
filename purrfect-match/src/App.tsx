import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { SuccessStoryPage } from "SuccessStoryPage";
import { ContactPage } from "ContactPage";
import { HomePage } from "HomePage/HomePage";
import { SearchPage } from "SearchPage";
import { NavigationMenu } from "NavigationMenu/NavigationMenu";
import Footer from "Footer/Footer";
import { ShelterHomePage } from "ShelterHome";

interface AppState {
    page: JSX.Element | null;
}

export class App extends React.Component<{}, AppState> {
    constructor(props: any) {
        super(props);
        this.state = {
            page: null
        };
    }

    public render() {
        return (
            <Router>
                <NavigationMenu />
                <div className="topBar" />
                <Switch>
                    <Route path="/contact">
                        <ContactPage />
                    </Route>
                    <Route path="/success">
                        <SuccessStoryPage />
                    </Route>
                    <Route path="/search">
                        <SearchPage />
                    </Route>
                    <Route path="/shelterHome">
                        <ShelterHomePage />
                    </Route>
                    <Route path="/">
                        <HomePage />
                    </Route>
                </Switch>
                <Footer />
            </Router>
        );
    }
}
