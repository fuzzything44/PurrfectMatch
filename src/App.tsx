import React from 'react';
import PurrfectMenu from './PurrfectMenu/PurrfectMenu';
import Card from './HomePage/Card';

import logo from './PurrfectMenu/logo.png';

class App extends React.Component {
  render() {
    let links = [
      { label: 'Success Stories', link: '#success-stories', active: true },
      { label: 'Contact Us', link: '#contact-us' },
    ];

    return (
      <div>
        <div className="container center">
          <PurrfectMenu links={links} logo={logo} />
        </div>

        <div className="Cards">
          <div className="CardGroup">
            <Card
              title="Cat"
              image={require('./images/Cat_1.png')} />
            <Card
              title="Dog"
              image={require('./images/Dog_1.png')} />
            <Card
              title="Other"
              image={require('./images/Hamster_1.png')} />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
