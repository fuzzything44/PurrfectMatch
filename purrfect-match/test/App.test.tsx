
import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { App } from 'App';

configure({ adapter: new Adapter() });

test("Renders correctly", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
});
