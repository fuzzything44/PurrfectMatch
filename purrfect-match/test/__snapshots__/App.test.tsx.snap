// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<BrowserRouter>
  <NavigationMenu />
  <div
    className="topBar"
  />
  <Switch>
    <Route
      path="/contact"
    >
      <ContactPage />
    </Route>
    <Route
      path="/success"
    >
      <SuccessStoryPage />
    </Route>
    <Route
      path="/search"
    >
      <SearchPage />
    </Route>
    <Route
      path="/"
    >
      <HomePage />
    </Route>
  </Switch>
</BrowserRouter>
`;
